### PR TITLE
[Windows] Cleanup Certbot pkg dir before installing to avoid dependencies conflicts

### LINKED ISSUE #8824: Windows installer doesn't properly overwrite itself
## My operating system is (include version):
Windows 2012 R2

## I installed Certbot with (snap, OS package manager, pip, certbot-auto, etc):

beta download from website https://dl.eff.org/certbot-beta-installer-win32.exe

## Certbot's behavior differed from what I expected because:
cryptography problem, but depending on actual version can be different

I use certbot only to generate certs, I install them myself on the webservers

Had iirc 1.12 version installed, downloaded and installed 1.14 (? if thats' the current one) -- run into cryptography error. 
If you had a different version then a different problem can be reported (i run some tests with a few older versions I had).

Basically it seems that in my case /pkgs subfolder contained both versions of cryptography (2.8 and 3.4.5 iirc), which triggered an exception when running certbot. 

TLDR -- it seems installing a newer beta version by installer may cause a conflict in python installation. 

In my case there was a relatively easy workaround: I deleted pkgs subfolder manually and run the installer again. After that it worked as expected.

### PR DESCRIPTION
**Pull request by adferrand**
Fixes #8824

This PR makes the installer first delete (if exist) the previous `pkg` directory in the Certbot installation in order to avoid dependencies conflicts when a new version of Certbot (with new versions of dependencies) is intaller other an existing one.

I took the simplest approach here, which is to delete specifically the directories known to create conflicts, instead of more complex approaches that involve to factor in some way the complete uninstaller logic. This is because the complexity added without a clear improvement does not worth it in my opinion. More specifically:
* factorizing in some way the uninstaller section in the NSIS template make the installer use any potential new logic of a new installation of Certbot instead of the one applying for the current installation, and may create unexpected errors during installation or at runtime
* calling the existing `uninstaller.exe` would be better, but I could not find a proper way to let NSIS wait for the actual end of the uninstall logic, and again may create unexpected errors during installation or at runtime

### CODE REVIEW

**Comment by bmw** [comment]
Did you see https://stackoverflow.com/questions/4676898/how-to-execute-an-nsis-uninstaller-from-within-an-another-nsis-installer-and-wai? There are some comments saying the most upvoted answer doesn't work, however, that exact line is included in the NSIS documentation linked there. If it works and there's a way to conditionally run the uninstaller only if it exists, is there any reason that solution shouldn't be preferred?

If we take this approach, I wonder if it might be worth suggesting the change upstream as a fix for https://github.com/takluyver/pynsist/issues/66. The maintainer suggested it as a possible yet inefficient option 5 years ago so I wonder if they'd accept it upstream so they can finally close that issue and we can maintain a smaller diff to our `.nsi` file.

**Comment by adferrand** [comment]
Yes I saw this stackoverflow proposition, and noticed as other did before that the NSIS installer returns immediately instead of waiting for the uninstall process to finish.

**Comment by bmw** [comment]
That's unfortunate. I confirmed this works though so if you add a changelog entry about this I think we can merge this.

File: certbot/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ Certbot adheres to [Semantic Versioning](https://semver.org/).
 ### Changed
 
 * DNS plugins based on lexicon now require dns-lexicon >= v3.1.0
+* Windows installer now cleans up old Certbot dependencies packages
```
Comment (bmw) [inline certbot/CHANGELOG.md:14]:
```suggestion
* Windows installer now cleans up old Certbot dependency packages
```

File: certbot/CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,8 @@ Certbot adheres to [Semantic Versioning](https://semver.org/).
...
+  before installing the new ones to avoid versions conflicts.
```
Comment (bmw) [inline certbot/CHANGELOG.md:15]:
```suggestion
  before installing the new ones to avoid version conflicts.
```

**Review by bmw — CHANGES_REQUESTED** [review]
Other than my minor suggested tweaks to the changelog, this LGTM!

**Review by bmw — APPROVED** [review]

**Pull request merged**